packages/menu/src/Menu.tsx

```diff
@@ -71,25 +88,64 @@ function Menu({
   usePortal = true,
   adjustOnMutation = false,
   shouldClose = () => true,
-  open,
-  setOpen,
+  open: controlledOpen,
+  setOpen: controlledSetOpen,
   children,
   className,
   refEl,
   trigger,
   ...rest
 }: MenuProps) {
-  const [uncontrolledOpen, setUncontrolledOpen] = useState(false);
+  const [focused, setFocused] = useState(0);
+
+  const refs: Array<HTMLElement> = [];
+  const menuitems: Array<React.ReactElement> = [];
+
+  const updateChildren = (
+    childset: React.ReactElement,
+  ): Array<React.ReactElement> => {
+    return React.Children.map(childset, child => {
+      if (isMenuItemElement(child)) {
+        menuitems.push(child);
+        return React.cloneElement(child, {
+          ref: (ref: HTMLElement) => refs.push(ref),
+        });
+      }
+
+      if (typeof child.props.children === 'object') {
+        return React.cloneElement(child, {
+          children: updateChildren(child.props.children),
+        });
+      }
+
+      return child;
+    });
+  };
+
+  const updatedChildren = updateChildren(children);
+
+  const enabledIndexes = menuitems.reduce(
```

Comment:
Menu accepts children that aren't MenuItem's right?

What happens to this functionality if that's the case?

Reply:
No longer using this, relying on refs now